pipeline_steps/common.groovy

```diff
@@ -439,24 +449,72 @@ def prepareRpcGit(String branch = "auto", String dest = "/opt"){
 
     print("Repo: ${env.RPC_REPO} Branch: ${branch}")
 
-    // checkout used instead of git as a custom refspec is required
-    // to checkout pull requests
-    checkout([$class: 'GitSCM',
-      branches: [[name: branch]],
-      doGenerateSubmoduleConfigurations: false,
-      extensions: [[$class: 'CleanCheckout']],
-      submoduleCfg: [],
-      userRemoteConfigs: [
-        [
-          url: env.RPC_REPO,
-          refspec: '+refs/pull/*:refs/remotes/origin/pr/* +refs/heads/*:refs/remotes/origin/*'
-        ]
-      ]
-    ]) // checkout
-    sh "git submodule update --init"
+    clone_with_pr_refs(env.RPC_REPO, branch)
   } // dir
 }
 
+// Clone repo with Refspecs required for PRs.
+// Should't need to supply any params to checkout a PR merged with the base.
+// Uses shell+git to avoid hostname verification failures with
+// the built in git scm step.
+// Use init + fetch instead of clone so that the repo can
+// be cloned into a non-empty directory. Thats added for
+// compatibility with the jenkins git scm step.
+void clone_with_pr_refs(
+  String repo="https://github.com/${env.ghprbGhRepository}.git",
+  String ref="origin/pr/${env.ghprbPullId}/merge",
+  String refspec='+refs/pull/\\*:refs/remotes/origin/pr/\\*'\
+                +' +refs/heads/\\*:refs/remotes/origin/\\*'
+){
```

Comment:
The neat thing about not passing the env vars is that you don't need to use them, for example pull id doesn't make sense for periodics, so a different structure for repo and ref can be used. See the usage in prepareRpcGit for example, sometimes ref is simply "master".